qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -67,13 +67,10 @@ def __post_init__(self):
 class CurveData:
     """Set of extracted experiment data."""
```

Comment:
I think this class should have a but more detail in the class doc since there is no init method to describe the args( ie what fields are and what they mean / do). Same with the other dataclasses in this file